app/src/main/java/org/simple/clinic/summary/linkId/LinkIdWithPatientSheetController.kt

```diff
@@ -25,7 +25,16 @@ class LinkIdWithPatientSheetController @Inject constructor(
     return Observable
         .merge(
             addIdToPatient(replayedEvents),
-            cancelAddingIdToPatient(replayedEvents))
+            cancelAddingIdToPatient(replayedEvents),
+            displayIdentifier(replayedEvents)
+        )
+  }
+
+  private fun displayIdentifier(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<LinkIdWithPatientSheetCreated>()
+        .map { it.identifier }
+        .map { identifier -> { ui: Ui -> ui.renderIdentifierText(identifier) } }
   }
```

Comment:
I think that the controller is not doing anything here except rendering the UI from an event which is sent from the view itself.
We could avoid all this and call the `renderIdentifierText` method from `onFinishInflate` of `LinkIdWithPatientView`.

Reply:
True, but I added this so that the `LinkIdWithPatientShown` event is sent to Heap for analytics. Since this is no longer an activity or a screen, the only way to send this event to Heap is via the controller.